client/src/main/java/io/pravega/client/stream/impl/TransactionInfoImpl.java

```diff
@@ -0,0 +1,32 @@
+/**
+ * Copyright Pravega Authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.pravega.client.stream.impl;
+
+import io.pravega.client.stream.Stream;
+import io.pravega.client.stream.Transaction;
+import io.pravega.client.stream.TransactionInfo;
+import lombok.Data;
+
+import java.util.UUID;
+
+@Data
```

Comment:
@Data will handle all @ToString, @EqualsAndHashCode, @Getter / @Setter  and @RequiredArgsConstructor. Please see if you need all these.

Reply:
**@Setter** will not work as all stream, transactionId and status are final. Other then setter we require all.